### PR TITLE
fix(`SelectMenuBuilder`): properly accept `SelectMenuOptionBuilder`s

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Fixes issue where djs builders wouldn't properly handle select menu option builders passed into a select menu builder.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/discord.js/src/structures/SelectMenuBuilder.js [hunk 1]
```diff
@@ -21,18 +21,31 @@ class SelectMenuBuilder extends BuildersSelectMenu {
...
 
+  /**
+   * Normalizes a select menu option emoji
+   * @param {SelectMenuOptionData|JSONEncodable<APISelectMenuOption>} selectMenuOption The option to normalize
+   * @returns {Array<SelectMenuOptionBuilder | APISelectMenuOption>}
+   * @private
+   */
+  static normalizeEmoji(selectMenuOption) {
+    if (isJSONEncodable(selectMenuOption)) {
+      return selectMenuOption;
```
Comment (almeidx) [inline packages/discord.js/src/structures/SelectMenuBuilder.js:32]: The emoji is not being normalized here

Comment (suneettipirneni) [reply packages/discord.js/src/structures/SelectMenuBuilder.js:32]: It doesn't need to be, for builder classes toJSON normalizes the emoji.

File: packages/discord.js/src/structures/SelectMenuBuilder.js [hunk 2]
```diff
@@ -21,18 +21,31 @@ class SelectMenuBuilder extends BuildersSelectMenu {
     );
   }
 
+  /**
+   * Normalizes a select menu option emoji
+   * @param {SelectMenuOptionData|JSONEncodable<APISelectMenuOption>} selectMenuOption The option to normalize
+   * @returns {Array<SelectMenuOptionBuilder | APISelectMenuOption>}
```
Comment (Jiralite) [inline packages/discord.js/src/structures/SelectMenuBuilder.js:27]: For consistency in the documentation:

```suggestion
   * @returns {Array<SelectMenuOptionBuilder|APISelectMenuOption>}
```

Comment (almeidx) [reply packages/discord.js/src/structures/SelectMenuBuilder.js:27]: `(SelectMenuOptionBuilder|APISelectMenuOption)[]` to be consistent with the majority of the documentation

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**